grails-app/services/com/unifina/service/EthereumIntegrationKeyService.groovy

```diff
@@ -121,13 +121,19 @@ class EthereumIntegrationKeyService {
 	}
 
 	SecUser getEthereumUser(String address) {
-		List<IntegrationKey> keys = IntegrationKey.findAll {
-			idInService == address && service in [IntegrationKey.Service.ETHEREUM, IntegrationKey.Service.ETHEREUM_ID]
+		IntegrationKey key = IntegrationKey.createCriteria().get {
+			'in'("service", [IntegrationKey.Service.ETHEREUM, IntegrationKey.Service.ETHEREUM_ID])
```

Comment:
could the reason be this "in"? At least that's different. You could use `or` instead for only two cases.

Also I think maybe it should only be ETHEREUM_ID since this is used for identifying a user. I think otherwise the same key could be added to another user as integration key?

Reply:
@jtakalai Could you try what happens on your machine when you add the third argument?

Reply:
Both ETHEREUM and ETHEREUM_ID